coala_quickstart/info_extraction/InfoExtractor.py

```diff
@@ -14,7 +16,16 @@ def __init__(self,
         :param project_directory: Absolute path to project directory in which
                                   the target files will be searched.
         """
-        self.target_globs = target_globs
+        target_files = self.retrieve_files(target_globs, project_directory)
+        for fname in target_files:
+            if not fnmatch(fname, self.supported_file_globs):
+                raise ValueError("The taraget file {} does not match the "
```

Comment:
It's a check `supported_file_globs` is class attribute which is same throughout the instances. If a different kind of `target_glob` is passed to the `InfoExtractor` class that also yields files that to not match `supported_files_globs`, error will be raised.